src/github.rs

```diff
@@ -40,15 +47,16 @@ impl ServiceFactory for GithubServiceFactory {
 
     fn create_service(&self, matches: &getopts::Matches) -> CreateServiceResult {
         match (
+            matches.opt_str("github-endpoint"),
             matches.opt_str("github-org"),
             matches.opt_str("github-username"),
             matches.opt_str("github-password"),
         ) {
-            (Some(org), Some(username), Some(password)) => CreateServiceResult::Service(Box::new(
-                GithubService{org: org, username: username, password: password}
+            (endpoint, Some(org), Some(username), Some(password)) => CreateServiceResult::Service(Box::new(
+                GithubService{endpoint: endpoint.unwrap_or(DEFAULT_ENDPOINT.to_string()), org: org, username: username, password: password}
             )),
-            (None, None, None) => CreateServiceResult::None,
-            (org, username, password) => {
+            (None, None, None, None) => CreateServiceResult::None,
+            (_endpoint, org, username, password) => {
```

Comment:
`_` seems to be more idiomatic in Rust.

On Thu, Aug 27, 2015 at 12:24 AM, David Reid <notifications@github.com>
wrote:

> In src/github.rs
> <https://github.com/alex/otp-cop/pull/5#discussion_r38061090>:
>
> >              )),
> > -            (None, None, None) => CreateServiceResult::None,
> > -            (org, username, password) => {
> > +            (None, None, None, None) => CreateServiceResult::None,
> > +            (_endpoint, org, username, password) => {
>
> It's just a habit I've developed from using other pattern match heavy
> languages where you might have multiple ignored parameters in a single
> match. I can change it to _ if you want.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alex/otp-cop/pull/5/files#r38061090>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084